layouts/company/single.html

```diff
@@ -102,20 +102,33 @@
         {{ T "cdb-help-needed" (dict "editUrl" (print ("suggest" | absURL) "?type=edit&for=cdb&slug=" .Params.slug) "newUrl" (print ("suggest" | absURL) "?type=new&for=cdb")) | safeHTML }}
     </div>
 
-    <!-- TODO: This is currently a very naive approach, mostly since we don't exactly have any data to do this better. We will want to improve this in the future. Oh yeah, and this feature seems to be very 'non-deterministic', at least under our parameters. Not sure if that is a good or a bad thing. -->
+    <!-- TODO: This is currently a very naive approach, mostly since we don't exactly have any data to do this better. We will want to improve this in the future. -->
     {{ if not .Site.Params.devMode }}
-    {{ $related := (where (.Site.RegularPages.Related .) "Type" "company") | first 5 }}
-    {{ with $related }}
-    <div id="related-companies">
-        <h2>{{ T "cdb-related-companies" }}</h2>
-        <!-- TODO: Nicer styling -->
-        <ul>
-            {{ range . }}
-            <li><a href="{{ .Permalink }}">{{ .Param "name" }}</a></li>
+        {{ $related_pages := slice }}
+
+        {{ $num_categories := (cond (isset .Params "categories") .Params.categories slice) | len}}
+        {{ $num_countries := (.Param "relevant-countries") | len}}
+
+        {{ range .Site.Pages }}
+            {{ $num_common_categories := intersect $.Params.categories .Params.categories | len }}
+            {{ $num_common_countries := intersect ($.Param "relevant-countries") (.Param "relevant-countries") | len }}
+            {{ if and (gt $num_common_categories 0) (gt $num_common_countries 0) (ne .Permalink $.Permalink) }}
+                {{ $score := div (add (div (float $num_common_categories) $num_categories) (div $num_common_countries $num_countries)) 2 }}
```

Comment:
It seems to me categories should be prioritized over countries, since having common countries is not really that useful for the users, is it?

Reply:
Well, I'm pretty sure that's what the previous algorithm did anyway. Feel free to propose a different scoring system.